charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml

```
@@ -261,6 +261,11 @@ spec:
   storage:
 {{ tpl (toYaml .Values.prometheus.prometheusSpec.storageSpec | indent 4) . }}
 {{- end }}
+{{- if semverCompare ">= 1.27.0" (include "kube-prometheus-stack.operator.kubeVersion" .) }}
+  persistentVolumeClaimRetentionPolicy:
+    whenDeleted: {{ ternary "Delete" "Retain" .Values.prometheusOperator.statefulSet.pvcDeleteOnStsDelete }}
+    whenScaled: {{ ternary "Delete" "Retain" .Values.prometheusOperator.statefulSet.pvcDeleteOnStsScale }}
+{{- end }}
```

What about

```suggestion
{{- with .Values.prometheus.prometheusSpec }}
  persistentVolumeClaimRetentionPolicy:
    {{- toYaml .Values.prometheus.prometheusSpec.scrapeConfigNamespaceSelector | nindent 4 }}
{{- end }}
```

and a empty `scrapeConfigNamespaceSelector: {}` by default? That should also works out of the box, doesn't require the version check.

It also keeps the existing name schema.

---

Hi @Woojciech 

Sorry, I did a mistake and correct by suggestion. I hope question 1 and 2 is now more clear. A user could put:

```yaml
prometheus:
  prometheusSpec:
    persistentVolumeClaimRetentionPolicy:
      whenDeleted: Delete
      whenScaled: Retain
```

into the values to change the defaults.

> Could you please clarify on how do we know that in this case version is 1.27.0+?

In the default values, this should be appear:

```yaml
prometheus:
  prometheusSpec:
    persistentVolumeClaimRetentionPolicy: {}
```

which means, empty by default and the whole block might be not set in the values.

Additionally,. I'm shifting the condition to the end-user. The end-user should only define the flags, if he is aware that he is using 1.27. 

Your great documentation inside the values yaml also mention that 1.27 is required. I do not see the additional benefit of adding an technical condition. I

---

No problem, thanks for such detailed response :) I implemented your suggestions, I left whenDeleted and whenScaled fields in YAML commented out for reference, both set to default value which is Retain, let me know if that is fine, or if I should remove it at all.

PS. the nice docs are not necessarily mine, they were copied from the prometheus chart, so kudos to prometheus maintainers :)